code/modules/xenoarchaeology/tools/xenoarchaeology_machines.dm

```diff
@@ -0,0 +1,323 @@
+			//Handle contents
+			if(move_inside)
+				register_contents(I)
+	else
+		return ..()
+
+/obj/machinery/xenoarchaeology_machine/attack_hand(mob/living/user)
+	. = ..()
+	empty_contents()
+
+/obj/machinery/xenoarchaeology_machine/proc/register_contents(atom/atom_target)
+	RegisterSignal(atom_target, COMSIG_PARENT_QDELETING, PROC_REF(unregister_contents))
+	RegisterSignal(atom_target, COMSIG_MOVABLE_MOVED, PROC_REF(unregister_contents))
+	held_contents += atom_target
+
+/obj/machinery/xenoarchaeology_machine/proc/unregister_contents(datum/source)
+	SIGNAL_HANDLER
+
+	held_contents -= source
+	UnregisterSignal(source, COMSIG_PARENT_QDELETING)
+	UnregisterSignal(source, COMSIG_MOVABLE_MOVED)
+
+/obj/machinery/xenoarchaeology_machine/proc/get_target()
+	return move_inside ? src : drop_location()
+
+/obj/machinery/xenoarchaeology_machine/proc/empty_contents(atom/movable/target, force)
+	if(target && (list(target) & held_contents || force))
+		target.forceMove(get_turf(src))
+		unregister_contents(target)
+		return
+	for(var/atom/movable/AM in held_contents)
+		AM.forceMove(get_turf(src))
+		unregister_contents(AM)
+
+//Circuitboard
+/obj/item/circuitboard/machine/xenoarchaeology_machine
+	name = "place holder (Machine Board)"
+	icon_state = "science"
+	build_path = /obj/machinery/xenoarchaeology_machine
+	req_components = list(
+		/obj/item/stack/ore/bluespace_crystal = 5,
+		/obj/item/stock_parts/matter_bin = 1)
+
+/*
+	Scale, measures artifact weight
+*/
+/obj/machinery/xenoarchaeology_machine/scale
+	name = "industrial scale"
+	desc = "A piece of industrial equipment, designed to weigh thousands of kilograms."
+	icon_state = "scale"
+	circuit = /obj/item/circuitboard/machine/xenoarchaeology_machine/scale
+
+/obj/machinery/xenoarchaeology_machine/scale/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Interact to measure artifact weight.\nLabeled artifacts will also show label weights, against the total.</span>"
+
+/obj/machinery/xenoarchaeology_machine/scale/attack_hand(mob/living/user)
+	. = ..()
+	///Get the combined weight of all artifacts in our target
+	var/atom/target = get_target()
+	var/total_weight = 0
+	var/label_weight = 0
+	for(var/atom/atom_target in target)
+		var/datum/component/xenoartifact/artifact_component = atom_target.GetComponent(/datum/component/xenoartifact)
+		if(artifact_component)
+			total_weight += artifact_component.get_material_weight()
+		//If there's a label and we're obliged to 'help' the player
+		var/obj/item/sticker/xenoartifact_label/label = locate(/obj/item/sticker/xenoartifact_label) in atom_target.contents
+		if(label)
+			for(var/datum/xenoartifact_trait/T as() in label.traits)
+				say("[initial(T.label_name)] - Weight: [initial(T.weight)]")
+				label_weight += initial(T.weight)
+		else if(isitem(atom_target) || isliving(atom_target))
+			if(isliving(atom_target))
+				if(prob(1))
+					say("Unexpected Fatass Detected!")
+					say("Get the fuck off me, lardass!")
```

Comment:
missing honking or farting sound :trollface:

Reply:
Hey, just want to apologize for making such a stupid fucking mistake. I know I'm a maint now, and I'm not supposed to miss such obvious things.